src/PeriodEnd/Refunds/SFA.DAS.ProviderPayments.Calc.Refunds/Services/LearnerBuilder.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Dto;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Infrastructure.Repositories;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Services.Dependencies;
+using SFA.DAS.ProviderPayments.Calc.Shared.Infrastructure.Data;
+
+namespace SFA.DAS.ProviderPayments.Calc.Refunds.Services
+{
+    public class LearnerBuilder : ILearnerBuilder
+    {
+        private readonly IRequiredPaymentRepository _requiredPaymentRepository;
+        private readonly IHistoricalPaymentsRepository _historicalPaymentsRepository;
+
+        private Dictionary<string, LearnerData> _learnerProcessParameters;
+       
+        public LearnerBuilder(
+            IRequiredPaymentRepository requiredPaymentRepository,
+            IHistoricalPaymentsRepository historicalPaymentsRepository)
+        {
+            _requiredPaymentRepository = requiredPaymentRepository;
+            _historicalPaymentsRepository = historicalPaymentsRepository;
+        }
+
+        public List<LearnerData> CreateLearnersForProvider(long ukprn)
+        {
+            ResetLearnerResultsList();
```

Comment:
Can you make this thread safe and make _learnerProcessParameters a local variable?